doc/lpc18/results.tex

```diff
@@ -1,35 +1,145 @@
 \section{Experimental results}\label{sec:results}
-
 \subsection{Testbed}
 All of our performance results use a hardware testbed that consists of
-two Intel Xeon E5 2440 servers, each with an Intel 10GbE X540-AT2 dual
+two Intel Xeon E5 2440 v2 1.9GHz servers, each with an Intel 10GbE X540-AT2 dual
 port NIC, with the two ports of the Intel NIC on one server connected
 to the two ports on the identical NIC on the other server.i
 We installed p4c-xdp on one server, the {\em target server}, and
 attached the XDP program to the port that receives the packets
 The other server, the {\em source server}, generates packets
 at the maximum 10~Gbps packet rate of 14.88~Mpps using the DPDK-based
 TRex~\cite{trex} traffic generator.  The source server sends minimum
-length 64-byte packets in various traffic patterns to one port of the
-target server, and receives the forwarded packets on the other port.
+length 64-byte packets in {\em single} UDP flow to one port of the
+target server, and receives the forwarded packets on the same port.
 At the target server, every packet received goes through the
 pipeline specified in P4.
-We measure the forwarding speed to evaluate the effectiveness of the
-generated XDP program.
 
-\subsection{Performance}
-We measured the sample programs under the tests directory.
+We use sample p4 programs under the tests directory and the following
+metrics to understand the performance impact of the P4-generated XDP
+program:
+\begin{itemize}
+\item Packet Processing Rate (Mpps): Once XDP program finishes processing
+the packet, it returns one of the actions mentioned in section~\ref{background}.
+We made a small modification to all p4 program to always return XDP\_DROP,
+so that we can count the number of packets being drop per second as a
+indication of how fast the XDP can process.
+\item CPU Utilization: Every packets processed by XDP program is run
+under the per-core software IRQ daemon, named ksoftirqd/<core id>.
+We measure the CPU utilization of the ksoftirqd on the core.
```

Comment:
Thanks for making changes based on my earlier request, but it still isn't clear to me whether there are multiple CPU cores involved in processing these packets in parallel, or at most one core at a time processing packets.

Do you know?

It is a pretty big difference if you have a 6-core machine with 6 cores running at 100% utilization, or a 6-core machine with 1 core running at 100% utilization, and 5 idle cores.

Reply:
Thanks!
it's only one core processing all packets. Let me revise to make it more clear.